File: PHP/0012.php
class Solution {

/**
 * @param Integer $num
 * @return String
 */
function intToRoman($num) {
    // Define the values and symbols
    $values = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1];
    $symbols = ["M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"];
    
    $result = "";
    
    // Iterate over the values and symbols
    for ($i = 0; $i < count($values); $i++) {
        // Determine how many times the current value fits into num
        while ($num >= $values[$i]) {
            // Append the corresponding symbol
            $result .= $symbols[$i];
            // Reduce num by the value
            $num -= $values[$i];
        }
    }
    
    return $result;
}
}

File: Ruby/0012.rb
# @param {Integer} num
# @return {String}
def int_to_roman(num)
    # Define the values and symbols
    values = [1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1]
    symbols = ["M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"]
  
    result = ""
  
    # Iterate over the values and symbols
    values.each_with_index do |value, index|
      while num >= value
        result += symbols[index]
        num -= value
      end
    end
  
    result
  end